plugins/xlua/scripts/A333.systems/A333.systems.lua

```diff
@@ -449,7 +448,38 @@ function auto_brake()
     end
 end
 
--- PFD STUFF
+-- AutoPilot Debug Tool
+
+local previous_ap_mode = -1
+
+function update()
+    local ap_mode = get("sim/cockpit/autopilot/autopilot_mode")
+
+    if ap_mode == 0 and previous_ap_mode ~= 0 then
+        local sim_time = get("sim/time/total_running_time_sec")
+        local reason = get("sim/cockpit2/autopilot/disconnect_reason")
+
+        local reason_text = "Unknown"
+
+        if reason == 1 then
+            reason_text = "Pilot input"
+        elseif reason == 2 then
+            reason_text = "Manual override"
+        elseif reason == 3 then
+            reason_text = "Trim error"
+        elseif reason == 4 then
+            reason_text = "Control limits exceeded"
+        end
+
+        logMsg("[A330 Debug] A/P Disconnected at sim time: " .. sim_time .. " | Reason: " .. reason_text)
```

Comment:
```suggestion
        logMsg("[AG330 Debug] A/P Disconnected at sim time: " .. sim_time .. " | Reason: " .. reason_text)
```
That is about the schema we have been using.

Furthermore, I'd add more information:
I would add the change of the AP mode and the raw reason value:
```suggestion
        logMsg("[AG330 Debug] A/P mode changed (" .. previous_ap_mode .. " -> " .. ap_mode .. ") at sim time: " .. sim_time .. " | Reason: " .. reason .. " - " .. reason_text)
```